File: leetcode/solutions/0155.MinStack/minStack_test.ts
import { assertEquals } from "https://deno.land/std/testing/asserts.ts";
import MinStack from "./MinStack.ts";

Deno.test("0155. Min Stack", () => {
  const minStack = new MinStack();
  minStack.push(-2);
  minStack.push(0);
  minStack.push(-3);
  assertEquals(minStack.getMin(), -3);

  minStack.pop();
  assertEquals(minStack.top(), 0);
  assertEquals(minStack.getMin(), -2);
});

File: leetcode/solutions/0155.MinStack/MinStack_test.go
package solution

import (
	"testing"
)

func Test_0155_MinStack(t *testing.T) {
	minStack := Constructor()
	minStack.Push(-2)
	minStack.Push(0)
	minStack.Push(-3)
	got := minStack.GetMin()
	want := -3
	if got != want {
		t.Errorf("got %#v want %#v", got, want)
	}
	minStack.Pop()
	got = minStack.Top()
	want = 0
	if got != want {
		t.Errorf("got %#v want %#v", got, want)
	}
	got = minStack.GetMin()
	want = -2
	if got != want {
		t.Errorf("got %#v want %#v", got, want)
	}
}
